bot/cogs/events.py

```diff
@@ -0,0 +1,152 @@
+            help_command = (self.bot.get_command("help"),)
+
+        if isinstance(e, BadArgument):
+            await ctx.send(f"Bad argument: {e}\n")
+            await ctx.invoke(*help_command)
+        elif isinstance(e, UserInputError):
+            await ctx.invoke(*help_command)
+        elif isinstance(e, NoPrivateMessage):
+            await ctx.send("Sorry, this command can't be used in a private message!")
+        elif isinstance(e, BotMissingPermissions):
+            await ctx.send(
+                f"Sorry, it looks like I don't have the permissions I need to do that.\n\n"
+                f"Here's what I'm missing: **{e.missing_perms}**"
+            )
+        elif isinstance(e, CommandInvokeError):
+            await ctx.send(
+                f"Sorry, an unexpected error occurred. Please let us know!\n\n```{e}```"
+            )
+            raise e.original
+        log.error(f"COMMAND ERROR: '{e}'")
+
+    async def on_ready(self):
+        users = []
+
+        for member in self.bot.get_guild(PYTHON_GUILD).members:  # type: Member
+            roles = [r.id for r in member.roles]  # type: List[int]
+
+            if OWNER_ROLE in roles:
+                users.append({
+                    "user_id": member.id,
+                    "role": OWNER_ROLE
+                })
+            elif ADMIN_ROLE in roles:
+                users.append({
+                    "user_id": member.id,
+                    "role": ADMIN_ROLE
+                })
+            elif MODERATOR_ROLE in roles:
+                users.append({
+                    "user_id": member.id,
+                    "role": MODERATOR_ROLE
+                })
+            elif DEVOPS_ROLE in roles:
+                users.append({
+                    "user_id": member.id,
+                    "role": DEVOPS_ROLE
+                })
+
+        if users:
+            log.debug(f"{len(users)} user roles updated")
+            data = await self.send_updated_users(*users)  # type: dict
+
+            if any(data.values()):
+                embed = Embed(
+                    title="User roles updated"
+                )
+
+                for key, value in data.items():
+                    if value:
+                        embed.add_field(
+                            name=key.title(), value=str(value)
+                        )
+
+                await self.bot.get_channel(DEVLOG_CHANNEL).send(
+                    embed=embed
+                )
+
+    async def on_member_update(self, before: Member, after: Member):
+        if before.roles == after.roles:
+            return
+
+        before_role_names = [role.name for role in before.roles]  # type: List[str]
+        after_role_names = [role.name for role in after.roles]  # type: List[str]
+        role_ids = [r.id for r in after.roles]  # type: List[int]
+
+        log.debug(f"{before.display_name} roles changing from {before_role_names} to {after_role_names}")
+
+        if OWNER_ROLE in role_ids:
```

Comment:
You do this process twice, you should make a function that does this. Don't repeat yourself 😉